node/bft-consensus/src/lib.rs

```diff
@@ -60,8 +62,67 @@ pub enum BftError {
     EyreReport(String),
 }
 
+fn base_path(dev: Option<u16>) -> PathBuf {
+    // Retrieve the starting directory.
+    match dev.is_some() {
+        // In development mode, the ledger is stored in the repository root directory.
+        true => match std::env::current_dir() {
+            Ok(current_dir) => current_dir,
+            _ => PathBuf::from(env!("CARGO_MANIFEST_DIR")),
+        },
+        // In production mode, the ledger is stored in the `~/.aleo/` directory.
```

Comment:
what about tests that might not directly use the `dev` option? then again, would they use the entire `Validator` :thinking:? just some food for thought

Reply:
Fair—we might want to revisit this when tackling the CLI structure 🤔